pkg/internal/testhelpers/containers.go

```diff
@@ -163,6 +164,8 @@ func WithDB(t testing.TB, DBName string, superuser SuperuserStatus, deferNode2Se
 	f(db, t, PgConnectURL(DBName, superuser))
 }
 
+var connMux sync.Mutex // Needs a mux lock at GetReadOnlyConnection otherwise a DATA RACE warning from go.
```

Comment:
This makes no sense. Can you provide the data race output?

Reply:
```shell
level=debug ts=2021-06-09T10:20:10.287Z caller=query_engine.go:35 msg="Lookback delta is zero, setting to default value" value=5m0s
=== RUN   TestPromQLSeriesEndpoint/test_series_endpoint
level=info ts=2021-06-09T10:20:10.290Z caller=throughput.go:52 msg="Samples write throughput" samples/sec=15003
==================
WARNING: DATA RACE
Write at 0x00c006e0e2b0 by goroutine 350:
  github.com/timescale/promscale/pkg/internal/testhelpers.GetReadOnlyConnection.func1()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/internal/testhelpers/containers.go:180 +0xd9
  github.com/jackc/pgx/v4/pgxpool.ConnectConfig.func1()
      /home/hsingh/go/pkg/mod/github.com/jackc/pgx/v4@v4.8.0/pgxpool/pool.go:176 +0x346
  github.com/jackc/puddle.(*Pool).constructResourceValue()
      /home/hsingh/go/pkg/mod/github.com/jackc/puddle@v1.1.1/pool.go:470 +0x743
  github.com/jackc/puddle.(*Pool).Acquire()
      /home/hsingh/go/pkg/mod/github.com/jackc/puddle@v1.1.1/pool.go:302 +0x704
  github.com/jackc/pgx/v4/pgxpool.(*Pool).Acquire()
      /home/hsingh/go/pkg/mod/github.com/jackc/pgx/v4@v4.8.0/pgxpool/pool.go:355 +0xab
  github.com/jackc/pgx/v4/pgxpool.(*Pool).Query()
      /home/hsingh/go/pkg/mod/github.com/jackc/pgx/v4@v4.8.0/pgxpool/pool.go:404 +0x6f
  github.com/timescale/promscale/pkg/pgxconn.(*connImpl).Query()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/pgxconn/pgx_conn.go:51 +0xc6
  github.com/timescale/promscale/pkg/pgmodel/querier.(*pgxQuerier).queryMultipleMetrics()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/pgmodel/querier/querier.go:216 +0x154
  github.com/timescale/promscale/pkg/pgmodel/querier.(*pgxQuerier).getResultRows()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/pgmodel/querier/querier.go:172 +0x55e
  github.com/timescale/promscale/pkg/pgmodel/querier.(*pgxQuerier).Select()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/pgmodel/querier/querier.go:75 +0xca
  github.com/timescale/promscale/pkg/query.querier.Select()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/query/queryable.go:57 +0x149
  github.com/timescale/promscale/pkg/query.(*querier).Select()
      <autogenerated>:1 +0x3c
  github.com/timescale/promscale/pkg/api.series.func1()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/api/series.go:79 +0x139c
  github.com/timescale/promscale/pkg/api.corsWrapper.func1()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/api/common.go:161 +0xab
  net/http.HandlerFunc.ServeHTTP()
      /home/hsingh/go/go/src/net/http/server.go:2069 +0x51
  github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1()
      /home/hsingh/go/pkg/mod/github.com/!n!y!times/gziphandler@v1.1.1/gzip.go:336 +0x45e
  net/http.HandlerFunc.ServeHTTP()
      /home/hsingh/go/go/src/net/http/server.go:2069 +0x51
  github.com/timescale/promscale/pkg/api.timeHandler.func1()
      /home/hsingh/go/src/github.com/timescale/promscale/pkg/api/router.go:150 +0xe5
  github.com/prometheus/common/route.(*Router).handle.func1()
      /home/hsingh/go/pkg/mod/github.com/prometheus/common@v0.21.0/route/route.go:83 +0x3d7
  github.com/julienschmidt/httprouter.(*Router).ServeHTTP()
      /home/hsingh/go/pkg/mod/github.com/julienschmidt/httprouter@v1.3.0/router.go:387 +0x1201
  github.com/prometheus/common/route.(*Router).ServeHTTP()
      /home/hsingh/go/pkg/mod/github.com/prometheus/common@v0.21.0/route/route.go:121 +0x69
  net/http.serverHandler.ServeHTTP()
      /home/hsingh/go/go/src/net/http/server.go:2887 +0xca
  net/http.(*conn).serve()
      /home/hsingh/go/go/src/net/http/server.go:1952 +0x87d
```

Reply:
But, if we have a lock, no race happens.

Reply:
In fact, I think this is after recent changes in postgres jack lib upstream. I got the error in `warning slow ingestion` PR as well. We don't see in our CIs as we don't apply `-race` flag in the master, which we should.
https://github.com/timescale/promscale/blob/66d8a94edf2078180d014c6a294c66ead632637d/.github/workflows/go.yml#L152